Pomodoro/Sources/App/ViewController.swift

```diff
@@ -6,15 +6,243 @@
+                .withTintColor(.red, renderingMode: .alwaysOriginal), for: .normal)
+        button.imageView?.contentMode = .scaleAspectFit
+        return button
+    }()
+
+    private lazy var stackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.alignment = .center
+        stackView.distribution = .equalSpacing
+
+        return stackView
+    }()
+
+    // MARK: - Lifecycle
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
-        view.backgroundColor = .red
+        view.backgroundColor = .systemBackground
+        drawBackgroundLayer()
+        setupStackView()
+        setupHierarchy()
+        setupLayout()
+    }
+
+    // MARK: - Setup
+
+    private func setupStackView() {
+        stackView.addArrangedSubview(modeLabel)
+        stackView.addArrangedSubview(timerLabel)
+        stackView.addArrangedSubview(startPauseButton)
+    }
+
+    private func setupHierarchy() {
+        view.addSubviews(stackView)
+    }
+
+    private func setupLayout() {
+        stackView.snp.makeConstraints { make in
+            make.centerX.equalTo(view)
+            make.top.equalToSuperview().offset(100)
+            make.bottom.equalToSuperview().offset(-100)
+        }
+    }
+
+    // MARK: - Actions
+
+    /// Background circle progress bar
+    private func drawBackgroundLayer() {
+        backgroundProgressLayer.path = UIBezierPath(
+            arcCenter: CGPoint(x: view.frame.midX, y: view.frame.midY),
+            radius: 150,
+            startAngle: -90.degreesToRadians,
+            endAngle: 270.degreesToRadians,
+            clockwise: true
+        ).cgPath
+        backgroundProgressLayer.strokeColor = UIColor.lightGray.cgColor
+        backgroundProgressLayer.fillColor = UIColor.clear.cgColor
+        backgroundProgressLayer.lineWidth = 10
+        view.layer.addSublayer(backgroundProgressLayer)
+    }
+
+    /// Foreground circle progress bar
+    private func drawForegroundLayer() {
+        let animationColor = isWorkTime ? UIColor.red : UIColor.green
+        foregroundProgressLayer.path = UIBezierPath(
+            arcCenter: CGPoint(x: view.frame.midX, y: view.frame.midY),
+            radius: 150,
+            startAngle: -90.degreesToRadians,
+            endAngle: 270.degreesToRadians,
+            clockwise: true
+        ).cgPath
+        foregroundProgressLayer.strokeColor = animationColor.cgColor
+        foregroundProgressLayer.fillColor = UIColor.clear.cgColor
+        foregroundProgressLayer.lineWidth = 8
+        view.layer.addSublayer(foregroundProgressLayer)
+    }
+
+    private func startResumeAnimation() {
+        !isAnimationStarted ? startAnimation() : resumeAnimation()
+    }
+
+    private func startAnimation() {
+        resetAnimation()
+        foregroundProgressLayer.strokeEnd = 0.0
+        animation.keyPath = "strokeEnd"
```

Comment:
It is bettor to create CABasicAnimation inside the scope of startAnimation func. It is not used outside of this func